app/src/main/java/com/concordia/mcga/utilities/pathfinding/PathFinderTile.java

```diff
@@ -1,36 +1,50 @@
 package com.concordia.mcga.utilities.pathfinding;
 
-import android.util.Log;
+import android.support.annotation.NonNull;
 
-import org.apache.commons.lang3.builder.EqualsBuilder;
-import org.apache.commons.lang3.builder.HashCodeBuilder;
-import org.json.JSONException;
-import org.json.JSONObject;
+/**
+ * Object acts as a wrapper for {@link IndoorMapTile}. Holds
+ * additional information about the tiles that are specifically used
+ * by the pathfinder algorithms
+ */
+public class PathFinderTile implements Comparable<PathFinderTile> {
 
-public class PathFinderTile {
+    enum Type {START, DESTINATION}
 
-    public static final PathFinderTile MAX_COST = new PathFinderTile(Integer.MIN_VALUE, Integer.MIN_VALUE);
+    public static final PathFinderTile MAX_COST = new PathFinderTile();
 
     static {
         MAX_COST.setDistFromEnd(Integer.MAX_VALUE);
     }
 
+    private IndoorMapTile indoorMapTile;
     private Type tileType;
     private int distFromStart;
     private int distFromEnd;
-    private int coordinateX;
-    private int coordinateY;
     private PathFinderTile parent;
 
-    public PathFinderTile(int x, int y) {
-        this.coordinateX = x;
-        this.coordinateY = y;
+    private PathFinderTile(){}
+
+    public PathFinderTile(IndoorMapTile indoorMapTile) {
+        this.indoorMapTile = indoorMapTile;
+    }
+
+    public Type getTileType() {
+        return tileType;
     }
 
     public void setTileType(Type tileType) {
         this.tileType = tileType;
     }
 
+    public IndoorMapTile getIndoorMapTile() {
+        return indoorMapTile;
+    }
+
+    public void setIndoorMapTile(IndoorMapTile indoorMapTile) {
+        this.indoorMapTile = indoorMapTile;
+    }
+
     public int getCost() {
```

Comment:
**This has a nasty 🐛 .**

getCost() method always returns the (constant) distance between the start and the destination tiles.
So applying the compareTo() on two tiles from the same grid will always say that they're the "same", because both have the same cost.

Imagine we have start at (0,0) and destination at (10,10),

Point A is at (2,3) distance from start is 5, distance from destination is 15, the sum of both is 20
Point B is at (5,5) distance from start is 10, distance from destination is 10, the sum of both is 20
Point C is at (9,9) distance from start is 18, distance from destination is 2, the sum of both is 20

Distance from (0,0) to (10,10) is 20 so the bug is self-evident, bugs caused by faulty compareTo() implementations have tragic consequences and are very hard to spot... For example the sorted property can't be used correctly  for objects of this class, as in a TreeSet, if this is broken...